### PR TITLE
Run check.yml on pull request instead of on push

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
It doesn't make that much sense to check modifications once they're in the main branch.

### CODE REVIEW

**Review by DavidGOrtega — COMMENTED** [review]
While I understand what you say I do not see a bad thing to check master after all the merges.

**Comment by 0x2b3bfa0** [comment]
Let's ask our official CI/CD consultant. 🔔  @casperdcl 😉  Does it make any sense to run checks on the main branch?

Given that our actions rely on several external and potentially flaky services, it would be nice to check the main branch version with a `cron` trigger, but running a double-check after merging doesn't seem especially appealing to me.

**Comment by casperdcl** [comment]
yeah this is a common problem. Basically the method we're following (no merge into master without testing) means we shouldn't test master at all. But everyone gets gastric butterflies at this thought. I'd propose:

1. test on PR
2. cron test on master (at least weekly if not daily)
3. [optional] no test on master push

Personally I only follow (3) if the tests are slow

**Comment by DavidGOrtega** [comment]
> but running a double-check after merging doesn't seem especially appealing to me.
> But everyone gets gastric butterflies at this thought

it depends always on how the PR's owner treat its PR.
 - Do they pull origin master before merge?

**Comment by DavidGOrtega** [comment]
> Personally I only follow (3) if the tests are slow

I do not think that is slow here either.
As I say its important to pull origin and test prior to merge but... thats not the case for every user, even within our team

**Comment by casperdcl** [comment]
> Do they pull origin master before merge?

we can disable merge-without-sync (last option below):

image

**Comment by 0x2b3bfa0** [comment]
> _we can prevent merge-without-sync_

:100: **That** gives me gastric butterflies, in a good way.

**Comment by DavidGOrtega** [comment]
> we can prevent merge-without-sync (last option below):

Indeed we should use the branch protection rules, we pay for it. Im used to not have them, thats why I ensure the test in master

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 name: Check CML GitHub Action
-on: [push]
+on:
+  push:
+    branches: master
```
Comment (0x2b3bfa0) [inline .github/workflows/check.yml:4]:
```suggestion
```

No longer required as per https://github.com/iterative/setup-cml/pull/37#issuecomment-911720323

**Review by DavidGOrtega — APPROVED** [review]
After branch protection lgtm

**Comment by 0x2b3bfa0** [comment]
@DavidGOrtega, people who approved https://github.com/iterative/setup-cml/pull/37 also approved https://github.com/iterative/setup-dvc/pull/27 😉 

552bf9bf2c70bd508249bbc9254d4f2b

**Pull request merged**